Feat/domain

Jmorales1306 opened the pull request:
## Description
En este PR se establece la base principal del sistema de reservas (Booking System) 

## Pull request checklist
Please check if your PR fulfills the following requirements:
- [X] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed. (Add screenshot)
- [ ] Is my code avoiding performance problems (ex: rendering issues, amount of queries against the database, n+1 problem, infinite loops, memory usage)
- [ ] Is my code handling integration exceptions (ex: timeout error while calling an external service)
- [ ] My code includes usage metrics (ex: google analytics events, logs)
- [ ] My code includes restriction of access based on roles and permissions


## Pull request type
<!-- Please do not submit updates to dependencies unless it fixes an issue. --> 
<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. --> 

Please check the type of change your PR introduces:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] Feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->


## Commands to be executed or config files to be changed

Jmorales1306 committed:
Add initial solution and project files for Booking System

Jmorales1306 committed:
Add entity classes for Booking System: Booking, Client, Location, Permission, Role, RolePermission, Room, and User

Jmorales1306 committed:
Add global using directive for Booking.Core.Entities

Jmorales1306 committed:
Add repository interfaces for Booking System: IBookingRepository, IClientRepository, ILocationRepository, IPermissionRepository, IRoleRepository, IRoomRepository, IUserRepository, and IRolePermission

Jmorales1306 committed:
Feat: implement Repository and Unit of Work design patterns

Jmorales1306 committed:
Fix: revert TargetFramework from net9.0 to net8.0 in Booking.Core.csproj

Jmorales1306 committed:
Refactor entity classes to use parameterless constructors and initialize properties in constructors

Jmorales1306 committed:
Refactor IBookingRepository: remove GetByReference method and add room availability checks

HamiltonG98 commented on src/Booking.Core/Interface/IRolePermission.cs:
La tabla pivote no deberia tener acciones, todo se deberia realizar desde Role o desde Permission

Jmorales1306 replied:
Listo, investigue un poco y lo que me recomendaste hacer es una manera de implementar DDD dentro de CleanArquitecture. Muchas gracias por la observacion.

Jmorales1306 committed:
git commit -m "Refactor(Core): Implement Aggregate Root pattern in Role

- Add AddPermission/RemovePermission methods to Role entity
- Remove IRolePermission interface
